core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -372,5 +372,5 @@ private[spark] object SparkConf {
   /**
    * Return whether the given config is a Spark port config.
    */
-  def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.endsWith(".port")
+  def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.contains(".port")
```

Comment:
you are misunderstanding what this method does. It looks for `spark.*.port` intentionally and should not match `spark.port.maxRetries`